addons/account/models/account_move_line.py

```diff
@@ -459,7 +459,7 @@ def _compute_currency_id(self):
     def _compute_name(self):
         for line in self:
             if line.display_type == 'payment_term':
-                if not line.name:
+                if not line.move_id.inalterable_hash:
```

Comment:
We actually never want to recompute when the move is hashed, even for product lines